ProcessLib/ComponentTransport/ComponentTransportFEM.h

```diff
@@ -706,41 +706,42 @@ class LocalAssemblerData : public ComponentTransportLocalAssemblerInterface
 
     std::vector<double> const& getIntPtDarcyVelocity(
         const double t,
-        GlobalVector const& current_solution,
-        NumLib::LocalToGlobalIndexMap const& dof_table,
+        std::vector<GlobalVector*> const& x,
+        std::vector<NumLib::LocalToGlobalIndexMap const*> const& dof_table,
         std::vector<double>& cache) const override
     {
-        auto const indices = NumLib::getIndices(_element.getID(), dof_table);
-        assert(!indices.empty());
+        assert(x.size() == dof_table.size());
 
-        if (_coupled_solutions == nullptr)  // monolithic scheme
+        auto const n_processes = x.size();
+        std::vector<std::vector<double>> local_xs;
+        local_xs.reserve(n_processes);
+
+        for (std::size_t process_id = 0; process_id < n_processes; ++process_id)
         {
-            auto const local_x = current_solution.get(indices);
+            auto const indices =
+                NumLib::getIndices(_element.getID(), *dof_table[process_id]);
+            assert(!indices.empty());
+            local_xs.push_back(x[process_id]->get(indices));
```

Comment:
`x` and `local_xs` stand for the global and local solution. For consistency in concept, I would name the local solution as `local_x`. What do you think?

Reply:
:heavy_check_mark:.